folly/Utility.h

```diff
@@ -384,4 +385,13 @@ constexpr auto to_unsigned(T const& t) -> typename std::make_unsigned<T>::type {
   return static_cast<U>(t);
 }
 
+template <class E>
+FOLLY_NODISCARD constexpr auto to_underlying_type(E e) noexcept
```

Comment:
I tend to prefer writing return types explicitly for public and exported functions.

The `[[no_discard]]` does not seem useful because this function does not have
* any status / error condition which are signaled by the return value
* any side-effects which are managed by the return value

Using `[[no_discard]]` on pure functions such as this does not seem to offer value to me.

Reply:
Thanks for your explanation. We've gone back and forth about our use of `[[no_discard]]` in our codebases. We began by sprinkling it on all free functions and member functions (shame we can't use it on lambdas...), but after a while of use, we came to the same reasoning you have: only to use it where it is most likely a true error if a caller ignores the return values (such as an `empty()` or `size()` member function on a container).

We're in agreement and I've removed the `[[no_discard]]` attribute.